src/rdb.c

```diff
@@ -2319,6 +2319,13 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid, int *error) {
                 }
                 o->type = OBJ_SET;
                 o->encoding = OBJ_ENCODING_LISTPACK;
+
+                if (setTypeSize(o) == 0) {
+                    zfree(encoded);
+                    o->ptr = NULL;
+                    decrRefCount(o);
+                    goto emptykey;
```

Comment:
rdbReportCorruptRDB
another option

Reply:
in #9297 we decided to silently ignore all of these in an RDB (and fail a RESTORE command).
that's why we'll goto `emptykey` rather than report corruption.